### PR TITLE
ref(py3): requirements followup, pin mock package for backported fixes from 3.8 stdlib mock

### PR DESCRIPTION
**Pull request by joshuarli**
Follow up to https://github.com/getsentry/sentry/pull/23245/files#r562843286.

I also manually reaudited `ImportError` for any stray py2 branching.

#sync-getsentry

### CODE REVIEW

**Comment by joshuarli** [comment]
Hmm, `tests/sentry/event_manager/test_event_manager.py` wfm locally. Gonna retry.

File: src/sentry/testutils/helpers/task_runner.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 from celery import current_app
 from contextlib import contextmanager
 from django.conf import settings
-from mock import patch
+from sentry.utils.compat.mock import patch
```
Comment (wedamija) [inline src/sentry/testutils/helpers/task_runner.py:8]: Is the idea that we'd remove this compat thing in another pr?

**Review by wedamija — APPROVED** [review]

File: src/sentry/testutils/helpers/task_runner.py (hunk 1 above)
Comment (joshuarli) [reply src/sentry/testutils/helpers/task_runner.py:8]: Yeah, it'd be a lot more modifications and I wanted to keep a followup small.

**Comment by joshuarli** [comment]
Ah, actually, some autofixes I made were bad and I didn't detect them, thanks tests.

**Comment by joshuarli** [comment]
Oh lol, something's up with mock_calls and I'm using 3.8 in my dev env, and it was fixed in 3.8:  https://bugs.python.org/issue37972

**Comment by joshuarli** [comment]
So I'm going to explicitly pin mock and change our compat mock to use mock instead of stdlib.

**Review by evanpurkhiser — APPROVED** [review]
nice

**Pull request merged**